### DISCUSSION TITLE
用视频纹理替换pag图层导致画面闪烁

### DISCUSSION THREAD

**Discussion by hjm1fb** [Tencent/libpag#2182] [Q&A]
我在接C++版PAG的时候，发现播放pag时，替换图层的方法mPagFile->replaceImage(index, replaceImage);每一帧渲染都要调用，不然有的pag文件会有PAG画面闪烁的问题。虽然replaceImage里的纹理ID和宽高尺寸都是不变的。替换纹理对应的输入源是图片或者视频，都有这个问题。pag里是一个转场的效果。pagFile和pagPlayer都只创建一次。
想问:
1. 闪烁的原因是什么
2. 目前解决闪烁的方式，就是每一帧都总是调用下面的函数来替换图层:
 auto replaceTexture = pag::BackendTexture(replaceGlInfo, tex->width(),
                                                          tex->height());
auto replaceImage = pag::PAGImage::FromTexture(replaceTexture, origin);
replaceImage->setScaleMode(mScaleModeForReplace);
mPagFile->replaceImage(index, replaceImage);
请问这个性能开销大么

**Comment by CodeJhF** [comment]
抱歉这条当时没有被跟进到，是我们漏了。

如果你还在使用 libpag 且问题仍然存在，麻烦回复一下当前版本号（4.4 / 4.5 系列）和最小复现 demo，我们会优先跟进；如果不再需要跟进，本条就先关闭了。（stale-close）
